### ISSUE TITLE
Exported Certain Roll20 Campaigns Causes Javascript Error - Uncaught (in promise) TypeError: string.toLocaleLowerCase is not a function

### ISSUE THREAD

**Issue by kiloforce** [kakaroto/R20Converter#84]
This doesn't appear to affect all Roll20 exported campaigns, but specifically at least affects a clean deployment/export/convert of Curse of Strahd and Dungeon of the Mad Mage campaigns.

This Javascript error appears in the Javascript console during the FoundryVTT load.  Not sure how bad this is or if this Javascript error could potentially break game mechanics, or other module functions, but any errors are always very concerning...

```
foundry.js:145 Uncaught (in promise) TypeError: string.toLocaleLowerCase is not a function
    at WordTree.addLeaf (foundry.js:145:21)
    at DocumentIndex._addLeaf (foundry.js:367:35)
    at DocumentIndex._indexCompendium (foundry.js:379:12)
    at foundry.js:276:14
    at new Promise (<anonymous>)
    at DocumentIndex.index (foundry.js:269:26)
```
![image](https://user-images.githubusercontent.com/433058/193437940-64dcd77c-32bc-4286-b648-d65aef25ee7d.png)

Converted to module with:
```
python3 /home/vtt/foundryconvert/R20Converter-0.11.6-linux/src/main.py --export-as-module --campaign-title=madmage /home/vtt/foundrydata/Data/modules/madmage madmage.zip
```

No other FoundryVTT modules loaded at the time of testing this.

Foundry v10.286.  Roll20Converter v0.11.6.  Roll20Exporter v0.8.1

**Comment by kakaroto** [comment]
I'm unsure about this one, as I've never seen it, but it looks like it's a Foundry error, which I don't think is R20Converter's fault. It **might** be that the crash is caused by data that isn't perfectly right in the world itself, but not only was the world migration done by Foundry itself (i.e: if something wasn't migrated to v10 properly, that'd still be a Foundry v10 issue) but also, it looks like it's an error when it tries to index compendiums, so if you have no modules enabled in your world, that's probably a problem with the dnd5e system's compendiums. 
Without a way to reliably reproduce the error, I can't quite say what's causing it, but I'd recommend filing the issue with Foundry since that's a problem on their side : https://github.com/foundryvtt/foundryvtt/issues
(If you find which 'bad data' is causing that crash, please report back, so if I can fix it in R20Converter, I'll do that, but Foundry core still needs to be made more robust against invalid data)

**Comment by kiloforce** [comment]
I think there might have been a miscommunication.  The *only* module that is activated, needed to produce this error, is one of the modules that Roll20Converter produced.  And only specifically on the larger campaigns like Curse of Strahd or Dungeon of the Mad Mage.

Steps for you to reproduce would be to:
1. Export one of the above mentioned campaigns using Roll20Exporter
2. Use Roll20Converter to generate a module
3. Activate the module in a new empty campaign.

Unfortunately this would require having access to one of the larger campaigns to export.

**Comment by kakaroto** [comment]
Ah ok, cause you're converting your campaign to a module instead of a world, I missed that info. 

I think that's likely because Foundry doesn't migrate data from compendiums until the user specifically tells it to.... using an API call... from console....

Unfortunately, this is how Foundry does things, breaking the API or database schema every few releases, which can cause this sort of headache. If you had converted to a world, the world would have been migrated automatically when you launched it at least.
Anyways, I've confirmed the issue with my curse of strahd game, I only see it on v10, and confirmed that if the packs are migrated properly, the error disappears.
Create a new macro in Foundry, set its type to `script`, then paste this code into it : 
```js
const moduleId = "curse-of-strahd";

const packs = game.packs.contents.filter(c => c.collection.startsWith(`${moduleId}.`));
for (const pack of packs) {
  await pack.configure({locked: false});
  await pack.migrate();
  await pack.configure({locked: true});
}
``` 

The first line specifies the id of the module (the folder name), it will then find any packs from that module, unlock them, then migrate its content. 
I confirmed that after running this script, the error disappears after I reload the page.
I really wish that Foundry was able to handle these types of migrations in a more user-friendly way.
